### ISSUE TITLE
Local files are not removed when using S3 object storage

### ISSUE THREAD

**Issue by jdaviescoates** [pixelfed/pixelfed#4114]
At least, this appears to be the case.

See e.g. this post https://bridport.pics/i/web/post/522792891553329155

After doing ``php artisan cache:clear`` right-clicking on the image in the above post takes me to the object storage version of the image (whereas beforehand it took me to the local version - although with my latest post right-clicking never took me to local storage), but the local version doesn't seem to be getting deleted from the local storage (and the whole point of me using S3 object storage is to avoid my local disk filling up).

**Comment by jdaviescoates** [comment]
Just to add, the issue here is really that seemingly it is not possible to have avatar/ profile images working when FILESYSTEM_DRIVER=s3 and so in order to get those to work (at least as far as I understand it), one currently **has** to use FILESYSTEM_DRIVER=local instead, but then all your data is stored twice, both locally and on s3.

If one uses FILESYSTEM_DRIVER=s3 then all data is only stored on s3, but avatars/ profile pics just don't work.

So really, in order to fix this problem, it would seem to me  (not being a coder and so not really knowing what I'm talking about, but just thinking out loud) that the simplest solution would be to make avatars/ profile pics work when FILESYSTEM_DRIVER=s3.

Thanks!

**Comment by mitexleo** [comment]
> Just to add, the issue here is really that seemingly it is not possible to have avatar/ profile images working when FILESYSTEM_DRIVER=s3 and so in order to get those to work (at least as far as I understand it), one currently **has** to use FILESYSTEM_DRIVER=local instead, but then all your data is stored twice, both locally and on s3.
> 
> If one uses FILESYSTEM_DRIVER=s3 then all data is only stored on s3, but avatars/ profile pics just don't work.
> 
> So really, in order to fix this problem, it would seem to me (not being a coder and so not really knowing what I'm talking about, but just thinking out load) that the simplest solution would be to make avatars/ profile pics work when FILESYSTEM_DRIVER=s3.
> 
> Thanks!

It should delete the media files from Local Disk automatically 12 hours (Only Local Posts) . It uploads the files to Local storage for optimization then upload again to s3.

The problem I'm currently having is with remote posts ... They're being stored twice ! Which is eating up my Local storage !

**Comment by jdaviescoates** [comment]
> It should delete the media files from Local Disk automatically 12 hours (Only Local Posts) . It uploads the files to Local storage for optimization then upload again to s3.

Thanks, perhaps I just didn't wait long enough.. let me investigate further...

> The problem I'm currently having is with remote posts ... They're being stored twice ! Which is eating up my Local storage !

To be clear, by "remote posts" I'm presuming you mean posts from other instances that are getting federated to your instance? And that those are not getting cleaned up after 12 hours like your local post? right?

**Comment by mitexleo** [comment]
> > It should delete the media files from Local Disk automatically 12 hours (Only Local Posts) . It uploads the files to Local storage for optimization then upload again to s3.
> 
> Thanks, perhaps I just didn't wait long enough.. let me investigate further...
> 
> > The problem I'm currently having is with remote posts ... They're being stored twice ! Which is eating up my Local storage !
> 
> To be clear, by "remote posts" I'm presuming you mean posts from other instances that are getting federated to your instance? And that those are not getting cleaned up after 12 hours like your local post? right?

Yeah .. Posts from other instances are not getting cleaned up from local storage..

**Comment by lapineige** [comment]
Does the command `php artisan media:migrate2cloud` fixes this ?
